src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -68,7 +68,16 @@ public function register_routes() {
 		// Lists/updates a single template based on the given id.
 		register_rest_route(
 			$this->namespace,
-			'/' . $this->rest_base . '/(?P<id>[\/\s%\w\.\(\)\[\]\@_\-]+)',
+			// The route.
+			sprintf(
+				'/%s/(?P<id>%s%s)',
+				$this->rest_base,
+				// Matches theme's directory: `/themes/<subdirectory>/<theme>/` or `/themes/<theme>/`.
+				// Excludes invalid directory name characters: `/:<>*?"|`.
+				'([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)',
+				// Matches the template name.
+				'[\/\w-]+'
+			),
```

Comment:
Before  [changeset 52376](https://core.trac.wordpress.org/changeset/52376), the regex was

```php
'/' . $this->rest_base . '/(?P<id>[\/\w-]+)',
```

The change restores the original regex but also adds theme matching, which is the same regex as in the global styles controller change above.